app/Sensor.php

```diff
@@ -72,15 +73,63 @@ public function getLatestDataAttribute()
     }
     
     /**
+     * Get the last hour's sensor data averaged by the minute for the sensor.
+     */
+    public function getLastHourMinutelyAvgDataAttribute()
+    {
+        return $this->hasMany('App\SensorData')
+            ->selectRaw("CONCAT(DATE_FORMAT(created_at, '%Y-%m-%d %H:%i'), ':00') AS date, AVG(value) AS value")
+            ->whereRaw("created_at > DATE_SUB(NOW(), INTERVAL 1 HOUR)")
+            ->groupBy(\DB::raw("DATE_FORMAT(created_at, '%Y-%m-%d %H:%i')"))
+            ->get();
+    }
+    
+    /**
      * Get the last day's sensor data averaged hourly for the sensor.
      */
     public function getLastDayHourlyAvgDataAttribute()
     {
-        // SELECT CONCAT(DATE_FORMAT(created_at, '%Y-%m-%d %H'), ':00:00') AS date, AVG(value) FROM sensor_data WHERE sensor_id = 20 AND created_at > DATE_SUB(NOW(), INTERVAL 1 DAY) GROUP BY DATE_FORMAT(created_at, '%Y-%m-%d %H')
-        return $this->hasOne('App\SensorData')
-            ->selectRaw("CONCAT(DATE_FORMAT(created_at, '%Y-%m-%d %H'), ':00:00') AS date, AVG(value)")
-            ->where('created_at', '>=', Carbon::now()->subDay())
-            ->groupByRaw("DATE_FORMAT(created_at, '%Y-%m-%d %H')")
-            ->get() ?? (object)[];
+        // SELECT CONCAT(DATE_FORMAT(created_at, '%Y-%m-%d %H'), ':00:00') AS date, AVG(value) As value FROM sensor_data WHERE sensor_id = 20 AND created_at > DATE_SUB(NOW(), INTERVAL 1 DAY) GROUP BY DATE_FORMAT(created_at, '%Y-%m-%d %H')
+        return $this->hasMany('App\SensorData')
+            ->selectRaw("CONCAT(DATE_FORMAT(created_at, '%Y-%m-%d %H'), ':00:00') AS date, AVG(value) AS value")
+            ->whereRaw("created_at > DATE_SUB(NOW(), INTERVAL 1 DAY)")
+            ->groupBy(\DB::raw("DATE_FORMAT(created_at, '%Y-%m-%d %H')"))
+            ->get();
+    }
+    
+    /**
+     * Get the last weeks's sensor data averaged daily for the sensor.
+     */
+    public function getLastWeekDailyAvgDataAttribute()
+    {
+        return $this->hasMany('App\SensorData')
+            ->selectRaw("CONCAT(DATE_FORMAT(created_at, '%Y-%m-%d '), '00:00:00') AS date, AVG(value) AS value")
+            ->whereRaw("created_at > DATE_SUB(NOW(), INTERVAL 7 DAY)")
+            ->groupBy(\DB::raw("DATE_FORMAT(created_at, '%Y-%m-%d ')"))
```

Comment:
Replace `groupBy(\DB::raw("DATE_FORMAT(created_at, '%Y-%m-%d ')"))` with `groupBy('date')` and you don't need to disable the MySQL strict setting. You should make this change for all of the other queries also.

Reply:
Also the `CONCAT()` isn't needed since all it is doing is adding `00:00:00` to every single point on your chart. I am not sure what you want but if you want to group by day you could do:
```php
$this->hasMany('App\SensorData')
            ->selectRaw("DATE_FORMAT(created_at, '%Y-%m-%d') AS date, AVG(value) AS value")
            ->whereRaw("created_at > DATE_SUB(NOW(), INTERVAL 7 DAY)")
            ->groupBy('date')
            ->get();
```

Reply:
Fixed in b481fe4, thanks!